packages/amplify-cli-extensibility-helper/src/types/api/amplify-api-resource-stack-types.ts

```diff
@@ -67,25 +66,25 @@ export interface OpenSearchDirectiveStack {
 }
 
 export interface FunctionDirectiveStack {
-  lambdaDataSource: Record<string, CfnDataSource>;
-  lambdaDataSourceRole: Record<string, CfnRole>;
-  lambdaDataSourceServiceRoleDefaultPolicy: Record<string, CfnPolicy>;
+  lambdaDataSource?: Record<string, CfnDataSource>;
+  lambdaDataSourceRole?: Record<string, CfnRole>;
+  lambdaDataSourceServiceRoleDefaultPolicy?: Record<string, CfnPolicy>;
 }
 
 export interface PredictionsDirectiveStack {
-  RekognitionDataSource: CfnDataSource;
-  RekognitionDataSourceServiceRole: CfnRole;
-  TranslateDataSource: CfnDataSource;
-  translateTextAccess: CfnPolicy;
-  LambdaDataSource: CfnDataSource;
-  LambdaDataSourceServiceRole: CfnRole;
-  LambdaDataSourceServiceRoleDefaultPolicy: CfnPolicy;
-  TranslateDataSourceServiceRole: CfnRole;
-  predictionsLambdaIAMRole: CfnRole;
-  predictionsLambdaFunction: CfnFunction;
-  PredictionsLambdaAccess: CfnRole;
-  predictionsIAMRole: CfnRole;
-  PredictionsStorageAccess: CfnPolicy;
-  identifyTextAccess: CfnPolicy;
-  identifyLabelsAccess: CfnPolicy;
+  RekognitionDataSource?: CfnDataSource;
```

Comment:
nit: if all the props will always be optional, it could be cleaner to write these types as:
```
export type PredictionsDirectiveStack = Partial<{
  ... stuff
}>;
```